src/app/hostname-generators/views/HostnameGeneratorDetailView.vue

```diff
@@ -0,0 +1,104 @@
+<template>
+  <RouteView
+    name="hostname-generator-detail-view"
+    :params="{
+      name: '',
+    }"
+    v-slot="{ route, t, uri }"
+  >
+    <RouteTitle
+      :title="t('hostname-generators.routes.items.title')"
+      :render="false"
+    />
+
+    <DataLoader
+      :src="uri(sources, '/hostnamegenerators/:name', {
+        name: route.params.name,
+      })"
+      v-slot="{ data }"
+    >
+      <AppView :docs="t('hostname-generators.href.docs')">
+        <template
+          v-if="data"
+          #title
+        >
+          <h1>
+            <XCopyButton
+              :text="data.name"
+            >
+              <RouteTitle
+                :title="t('hostname-generators.routes.item.title', { name: data.name })"
+              />
+            </XCopyButton>
+          </h1>
+        </template>
+        <div class="stack">
+          <AppAboutSection
+            :title="t('hostname-generators.routes.item.subtitle', { name: data.name })"
+            :created="t('common.formats.datetime', { value: Date.parse(data.creationTime) })"
+            :modified="t('common.formats.datetime', { value: Date.parse(data.modificationTime) })"
+          >
+            <div class="columns">
+              <DefinitionCard>
+                <template #title>
+                  {{ t('hostname-generators.common.selector') }}
+                </template>
+
+                <template
+                  v-if="data.selector"
+                  #body
+                >
+                  <XAction
+                    :to="{
+                      name: data.selector?.routeName,
+                      params: {
+                        mesh: data.mesh,
+                      },
+                    }"
+                  >
+                    {{ data.selector?.label }}
+                  </XAction>
```

Comment:
nit: If selector is definitely set from what I remember from your data transformer above both routeName and label are guaranteed to be set? Actually this made me realise something else (we should avoid having routeNames in the data layer), I made a comment above on that.

Reply:
I've addressed this with the changes related to your suggestions regarding the routeNames in data layer 🙂